### PR TITLE
R1.0

### PR DESCRIPTION
**Pull request by hanleycdavid**

### CODE REVIEW

**Review by pborysewicz — APPROVED** [review]
approved

File: Group6PublicLibrary/src/group6ranger/RangeSensor.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
+package group6ranger;
```
Comment (caw13) [inline Group6PublicLibrary/src/group6ranger/RangeSensor.java:1]: Again make sure you rename your packages to follow good Java package naming convention in your public repo as well.

File: Group6PublicLibrary/src/group6ranger/RangeSensor.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import java.util.*;
+import org.iot.raspberry.grovepi.GrovePi;
+import org.iot.raspberry.grovepi.devices.GroveUltrasonicRanger;
+
+/**
+ * Abstract subject provides methods for storing, removing, and notifying observers
+ */
+public abstract class RangeSensor{
+    private List<IRangeObserver> observers = new ArrayList<IRangeObserver>();
+    GroveUltrasonicRanger ranger;
```
Comment (caw13) [inline Group6PublicLibrary/src/group6ranger/RangeSensor.java:12]: Should be private or protected, if elements need access to it beyond these contexts they should be doing so via accessors, but only if necessary

File: Group6PublicLibrary/src/group6ranger/RangeSensor.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    @Override 
+    public String toString() {
+        return String.format("%s %s %s", this.getClass().getSimpleName(), grovePi, pin);
+    }
+    
+    /*
+     * An iterator over a RangeSensor's observers.
+     * @return an iterator that moves through the RangeSensor's observers in reverse order starting with the most recently attached observer.
+     */
+    public Iterator<IRangeObserver> iterator() {
```
Comment (caw13) [inline Group6PublicLibrary/src/group6ranger/RangeSensor.java:106]: Good use of Iterator pattern

File: Group6PublicLibrary/src/group6ranger/RangeSensor.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        return new RangeObserverIterator(this.observers);
+    }
+    
+    /*
+     * An iterator over a RangeSensor's observers.
+     *
+     * The RangeObserverIterator iterates over a RangeSensor's observers in reverse order starting with the most recently attached observer.
+     */
+    private class RangeObserverIterator implements Iterator<IRangeObserver> {
```
Comment (caw13) [inline Group6PublicLibrary/src/group6ranger/RangeSensor.java:115]: Good inner class use and iterator implementation

**Pull request merged**